lib/MultiCompiler.js

```diff
@@ -109,50 +155,54 @@ module.exports = class MultiCompiler extends Tapable {
 		let watchings = [];
 		let allStats = this.compilers.map(() => null);
 		let compilerStatus = this.compilers.map(() => false);
-		this.runWithDependencies(this.compilers, (compiler, callback) => {
-			const compilerIdx = this.compilers.indexOf(compiler);
-			let firstRun = true;
-			let watching = compiler.watch(Array.isArray(watchOptions) ? watchOptions[compilerIdx] : watchOptions, (err, stats) => {
-				if(err)
-					handler(err);
-				if(stats) {
-					allStats[compilerIdx] = stats;
-					compilerStatus[compilerIdx] = "new";
-					if(compilerStatus.every(Boolean)) {
-						const freshStats = allStats.filter((s, idx) => {
-							return compilerStatus[idx] === "new";
-						});
-						compilerStatus.fill(true);
-						const multiStats = new MultiStats(freshStats);
-						handler(null, multiStats);
+		if(this.validateDependencies(handler)) {
+			this.runWithDependencies(this.compilers, (compiler, callback) => {
+				const compilerIdx = this.compilers.indexOf(compiler);
+				let firstRun = true;
+				let watching = compiler.watch(Array.isArray(watchOptions) ? watchOptions[compilerIdx] : watchOptions, (err, stats) => {
+					if(err)
+						handler(err);
+					if(stats) {
+						allStats[compilerIdx] = stats;
+						compilerStatus[compilerIdx] = "new";
+						if(compilerStatus.every(Boolean)) {
+							const freshStats = allStats.filter((s, idx) => {
+								return compilerStatus[idx] === "new";
+							});
+							compilerStatus.fill(true);
+							const multiStats = new MultiStats(freshStats);
+							handler(null, multiStats);
+						}
 					}
-				}
-				if(firstRun && !err) {
-					firstRun = false;
-					callback();
-				}
+					if(firstRun && !err) {
+						firstRun = false;
+						callback();
+					}
+				});
+				watchings.push(watching);
+			}, () => {
+				// ignore
 			});
-			watchings.push(watching);
-		}, () => {
-			// ignore
-		});
+		}
 
 		return new MultiWatching(watchings, this);
 	}
 
 	run(callback) {
 		const allStats = this.compilers.map(() => null);
-		this.runWithDependencies(this.compilers, ((compiler, callback) => {
-			const compilerIdx = this.compilers.indexOf(compiler);
-			compiler.run((err, stats) => {
+		if(this.validateDependencies(callback)) {
```

Comment:
I guess this validation could be omitted as `runWithDependencies` do the validation

Reply:
I had it that way at first but compiler errors are [ignored](https://github.com/webpack/webpack/blob/622675993d72c6ea7d70cd2e503fd9416f940b2b/lib/MultiCompiler.js#L137) by the watcher. Since this is a configuration error that cannot be recovered should it report the error and exit?